### PR TITLE
Add connection events target

### PR DESCRIPTION
**Pull request by shbatm**
Add connection events for the Event Stream to allow subscription and callbacks. Attach a callback with `isy.connection_events(callback)` and receive a string with the event detail. See `constants.py` for events starting with prefix `ES_`.

This will allow detection and response in a subscriber to the different events, rather than just relying on the logs only.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "python.linting.pylintEnabled": true,
+    "python.formatting.provider": "black"
+}
```
Comment (OverloadUT) [inline .vscode/settings.json:4]: Should this be checked in? Normally I think IDE configs are ignored, but there are potential benefits

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
+### [v2.0.3] - Minor Updates, Connection Events and Timestamps
 
-#### Summary: 
+#### Added
+
+- Add connection events for the Event Stream to allow subscription and callbacks. Attach a callback with `isy.connection_events(callback)` and receive a string with the event detail. See `constants.py` for events starting with prefix `ES_`.
+
+### [v2.0.2] - Version 2.0 Initial Release
+
+#### Summary:
```
Comment (OverloadUT) [inline CHANGELOG.md:11]: Thanks for cleaning up the release notes to handle the 2.0.2 release kerfuffle

File: pyisy/isy.py [hunk 3]
```diff
@@ -149,11 +157,13 @@ def _auto_reconnecter(self):
                 self, self.conn.connection_info, self._on_lost_event_stream
             )
             self._events.running = True
+            self.connection_events.notify("reconnecting")
 
         if not self.auto_update:
             del self._events
             self._events = None
             self.log.warning("PyISY could not reconnect to the event stream.")
+            self.connection_events.notify("reconnect_failed")
```
Comment (OverloadUT) [inline pyisy/isy.py:166]: These should be the consts right?

File: .vscode/settings.json (hunk 1 above)
Comment (shbatm) [reply .vscode/settings.json:4]: Let's ignore this until we get a consistent working one we like. I think I added it to .gitignore on one of the other branches.

File: CHANGELOG.md (hunk 2 above)
Comment (shbatm) [reply CHANGELOG.md:11]: NP. I know the pain. Went through it on -beta when I originally changed everything.

File: pyisy/isy.py (hunk 3 above)
Comment (shbatm) [reply pyisy/isy.py]: Yep. Missed them after testing.

**Comment by shbatm** [comment]
Thanks for reviewing. I have 1 more PR I need to send later tonight before this batch can be set as a release...

It's actually a better fix for #98 that I found last night.

**Comment by OverloadUT** [comment]
Any preference in order of merging with these new PRs?

**Comment by shbatm** [comment]
Nope, whichever you pull first I'll rebase if needed. They should be pretty independent.

**Pull request merged**